include/revng/ABI/FunctionType/Layout.h

```diff
@@ -61,6 +61,9 @@ struct Layout {
   public:
     std::optional<StackSpan> Stack;
     ArgumentKind::Values Kind;
+
+    std::string Name;
+    std::string_view Comment;
```

Comment:
1. Why `std::string` vs `std::string_view`?
2. I'd stick with `StringRef` over `std::string_view`, unless we make `std::string_view` preferable over `StringRef` project-wide.

Reply:
1. `std::string` for name because that's what `name()` returns - hence we, as the caller, can make no assumption about the lifetime of it, while for the comment we use the explicit accessor (`Comment()`), hence we know that the original object owns the string - hence only a view is enough.

2. Sounds good. The only qualm with `StringRef` I have is that it's `llvm`, so when some part of the codebase doesn't need to explicitly depend on it, I prefer `std::string_view`, but I guess we're way too deep inside the llvm ecosystem to worry about stuff like that 😅.

Reply:
1. Gotcha
2. Eh yes. We abandoned `llvm::Optional` in favor of `std::optional`, one day we'll do the same for `string_view`, but mixing is counterproductive. So I'd ban `string_view` until further notice.

Reply:
This is gone.

Also, I replaced `std::string_view` with `llvm::StringRef` everywhere I noticed them.